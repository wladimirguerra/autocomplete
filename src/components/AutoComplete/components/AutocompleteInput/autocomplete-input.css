.autocomplete-input {
    display: flex;
    border: 2px solid;
    padding: 0;
}
