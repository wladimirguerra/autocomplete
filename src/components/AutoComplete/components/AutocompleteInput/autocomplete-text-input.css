.autocomplete-text-input {
    border: none !important;
    background-color: transparent;
    border-radius: 8px;
    padding: 8px;
    width: 100%;
}

.autocomplete-text-input:focus {
    border: none;
    outline: none;
}
