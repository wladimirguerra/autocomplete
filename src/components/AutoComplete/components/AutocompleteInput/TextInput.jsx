import React from "react";
import { theme } from "../../../../theme";
import "./autocomplete-text-input.css"

const inputStyle = {
    padding: theme.spacingUnit,
};

class TextInput extends React.Component {
    render() {
        const { forwardedRef, ...inputProps} = this.props

        return (
            <input style={inputStyle} className={"autocomplete-text-input"}  {...inputProps} ref={this.props.forwardedRef} type={"text"}/>
        );
    }
}

export default React.forwardRef((props, ref) => <TextInput {...props} forwardedRef={ref}/>) ;
