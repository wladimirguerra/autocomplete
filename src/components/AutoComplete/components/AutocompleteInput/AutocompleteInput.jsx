import * as React from "react";
import PropTypes from "prop-types";
import { theme } from "../../../../theme";
import TextInput from "./TextInput";
import { Loading } from "../Loading/Loading";
import "./autocomplete-input.css";

const rootStyle = {
    backgroundColor: theme.backgroundColor,
    borderColor: theme.primaryColor,
    borderRadius: theme.borderRadius,
};

const statusContainerStyle = {};

const inputContainerStyle = {
    flex: 1,
};

class AutocompleteInput extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            backgroundColor: theme.backgroundColor,
        };
    }

    handleFocusIn = () => {
        this.setState({
            backgroundColor: theme.focusedBackgroundColor,
        });
    };

    handleFocusOut = () => {
        this.setState({
            backgroundColor: theme.backgroundColor,
        });
    };

    render() {
        const { isLoading, inputRef, ...inputProps } = this.props;

        return (
            <div ref={this.props.inputRef} className={"autocomplete-input"} style={{ ...rootStyle, backgroundColor: this.state.backgroundColor }}>
                <div style={inputContainerStyle}>
                    <TextInput
                        {...inputProps}
                        onChange={this.props.onChange}
                        onFocus={this.handleFocusIn}
                        onBlur={this.handleFocusOut}
                        data-testid="autocomplete-input-container"
                    />
                </div>
                <div style={{ ...statusContainerStyle, visibility: isLoading ? "visible" : "hidden" }}>
                    <Loading/>
                </div>
            </div>
        );
    }
}

AutocompleteInput.propTypes = {
    isLoading: PropTypes.bool,
    onChange: PropTypes.func,
    value: PropTypes.string,
    name: PropTypes.string
};

export default React.forwardRef((props, ref) => <AutocompleteInput {...props} inputRef={ref}/>);
