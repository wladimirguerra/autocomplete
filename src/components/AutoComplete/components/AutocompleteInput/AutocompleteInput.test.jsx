import renderer from "react-test-renderer";
import AutocompleteInput from "./AutocompleteInput";
import { fireEvent, render } from "@testing-library/react";

test("AutocompleteInput renders correctly", () => {
    const tree = renderer.create(<AutocompleteInput/>).toJSON();

    expect(tree).toMatchSnapshot();
});

test("AutocompleteInput loading signal render correctly", () => {
    const tree = renderer.create(<AutocompleteInput isLoading={true}/>).toJSON();

    expect(tree).toMatchSnapshot();
});

test("AutocompleteInput is focused", () => {
    const { getByTestId, asFragment } = render(<AutocompleteInput/>);
    const input = getByTestId("autocomplete-input-container");

    fireEvent.focus(input);

    expect(asFragment()).toMatchSnapshot();
});
