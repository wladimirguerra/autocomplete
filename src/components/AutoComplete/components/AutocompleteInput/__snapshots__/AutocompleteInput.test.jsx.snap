// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteInput is focused 1`] = `
<DocumentFragment>
  <div
    class="autocomplete-input"
    style="background-color: rgb(228, 255, 226); border-color: #13bb00; border-radius: 8px;"
  >
    <div
      style="flex: 1;"
    >
      <input
        class="autocomplete-text-input"
        data-testid="autocomplete-input-container"
        style="padding: 8px;"
        type="text"
      />
    </div>
    <div
      style="visibility: hidden;"
    >
      <div
        class="loading"
        style="background-color: rgb(224, 130, 23); margin: 8px; height: 16px; width: 16px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`AutocompleteInput loading signal render correctly 1`] = `
<div
  className="autocomplete-input"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#13bb00",
      "borderRadius": 8,
    }
  }
>
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <input
      className="autocomplete-text-input"
      data-testid="autocomplete-input-container"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "padding": 8,
        }
      }
      type="text"
    />
  </div>
  <div
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    <div
      className="loading"
      style={
        Object {
          "backgroundColor": "#e08217",
          "height": 16,
          "margin": 8,
          "width": 16,
        }
      }
    />
  </div>
</div>
`;

exports[`AutocompleteInput renders correctly 1`] = `
<div
  className="autocomplete-input"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#13bb00",
      "borderRadius": 8,
    }
  }
>
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <input
      className="autocomplete-text-input"
      data-testid="autocomplete-input-container"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "padding": 8,
        }
      }
      type="text"
    />
  </div>
  <div
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    <div
      className="loading"
      style={
        Object {
          "backgroundColor": "#e08217",
          "height": 16,
          "margin": 8,
          "width": 16,
        }
      }
    />
  </div>
</div>
`;
